From 8265225b54e25aa978e0c4b8b1cf6e8cf589059d Mon Sep 17 00:00:00 2001
From: Brad Love <hidden@email.co>
Date: Tue, 8 Sep 2020 16:14:56 -0500
Subject: [PATCH 06/14] si2168: Include b 4.0-11 when checking firmware version

Multiple reports of firmware B-4.0-11 being non functional after resuming from sleep

Signed-off-by: Brad Love <hidden@email.co>
---
 drivers/media/dvb-frontends/si2168.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/media/dvb-frontends/si2168.c b/drivers/media/dvb-frontends/si2168.c
index fc05a2f92..b394448cf 100644
--- a/drivers/media/dvb-frontends/si2168.c
+++ b/drivers/media/dvb-frontends/si2168.c
@@ -631,8 +631,8 @@ static int si2168_sleep(struct dvb_frontend *fe)
 	if (ret)
 		goto err;
 
-	/* Firmware later than B 4.0-11 loses warm state during sleep */
-	if (dev->version > ('B' << 24 | 4 << 16 | 0 << 8 | 11 << 0))
+	/* Firmware B 4.0-11 and later lose warm state during sleep */
+	if (dev->version >= ('B' << 24 | 4 << 16 | 0 << 8 | 11 << 0))
 		dev->warm = false;
 
 	cmd_init(&cmd, "\x13", 1, 0);
-- 
2.43.0

